### DISCUSSION TITLE
Level generator not removing terrain on paths

### DISCUSSION THREAD

**Discussion by UnhappyWithMyUsername** [FakeFishGames/Barotrauma#13632] [Bug reports]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [X] My issue happened while using mods.

### What happened?

On DynamicEuropa I've had several reports about and experienced this on 1 occasion myself where the level generator doesn't remove the terrain on the generated path, thus possibly preventing the player from reaching the level destination or forcing them to go through the abyss as the only option. 

On the example screenshot provided to me you can see the mainpath clearly goes up through the terrain, yet doesn't clear the path for traversal. The only way was to go through the abyss and hope for another hole to the abyss somewhere in the level as the forced hole generated right at the start as indicated by the blue line. This was another users experience.

My experience was a seemingly normal level with a clear tunnel path which just didn't finish generating to the next outpost. Everything seemed fine until we got close to the destination outpost (about 70-80% of the way) and were suddenly in front of a wall of terrain. We turned around and found a different tunnel further down which turned out to be a dead end as well. After retrying the level multiple times we decided to take a look with freecam and noticed there was no way towards the outpost at all as the hole to the abyss generated at the destination outpost behind the terrain wall as well.

I suspect the level generation can break this way if the generator decides to generate the hole to the abyss at either the starting or ending location.

I specifically don't use the "CreateHoleNextToEnd" functionality as it always generates the hole on the level ceiling, which I don't want to happen since that may cause the destination outpost/exit to also generate at the ceiling, even if set to generate at a lower height as it always tries to adjusts to surrounding terrain above it. (Maybe make CreateHoleNextToEnd just create a hole horizontally at the same height as the destination is configured instead of always reaching for the ceiling height?)

Here's the screenshot which shows the first example mentioned:
![fFUS9FbLRFWorsHVUQltzw](https://github.com/Regalis11/Barotrauma/assets/74098498/d5c91a8a-720e-496f-9052-c094dcf5325e)

I sadly don't have a savegame with this occurrence at hand, but I will link this post on the DynamicEuropa page and inform people who experience this to provide the savegame and used modlist here.

### Reproduction steps

Play campaign with DynamicEuropa enabled.
Compatibility patches or level size patches don't matter. Reports indicate it happens with and without any of them.

### Bug prevalence

Happens every now and then

### Single player or multiplayer?

Other

### -

Unsure if only in MP or if SP is also affected. Happened to myself while hosting in-game (listen server)
Edit: Can now confirm it happens in both SP and MP.

### Version

v1.0.21.0

### Which operating system did you encounter this bug on?

Windows

**Comment by pavel4win** [comment]
It happens in multiplayer game.
When I trying to reach [название]горнодобывающий объект 
from station where save game is, there is no way.
It looks like the target is inside the rock.

[20230912-can-not-pass.zip](https://github.com/Regalis11/Barotrauma/files/12605134/20230912-can-not-pass.zip)

**Comment by BlakeQuiet** [comment]
On my way to Torpal Workshop, oh wait, no I'm not. 
Check this out; also there is no "Modlists" in my folder so I can't post that..
[Rip Ship [Dynamic Europa].zip](https://github.com/FakeFishGames/Barotrauma/files/15229769/Rip.Ship.Dynamic.Europa.zip)

**Comment by UnhappyWithMyUsername** [comment]
I finally got the issue to happen again to myself and keep the save at the exact point where it happens.
Requires quite a few mods to load so I made a steam collection and provided a link to it and everything else needed in the zip file.
Hopefully this helps debug the issue further.
[#NoPathAtOutpostSave.zip](https://github.com/user-attachments/files/16646304/NoPathAtOutpostSave.zip)

**Reply by SomeRandomNoobKekeke** [reply]
used:
-   `<Workshop name="DynamicEuropa - Frith's Biomes Compatibility Patch" id="2911529819" />`
-   `<Workshop name="DynamicEuropa" id="2532991202" />`
-  `<Workshop name="Frith's Biomes" id="2020332046" />`

and got spawned in wall
![20240817230857_1](https://github.com/user-attachments/assets/b64dbdce-8282-4fac-a622-31c20987f316)
also:
![123](https://github.com/user-attachments/assets/ca36213e-a76c-43e1-ae9f-018cfe5dfa28)

**Reply by UnhappyWithMyUsername** [reply]
Good to know only these mods are needed to get the same results 👍 
And yeah debug view makes the game shit itself in that level.
